src/goto-instrument/goto_instrument_parse_options.cpp

```
@@ -641,15 +642,49 @@ int goto_instrument_parse_optionst::doit()
 
     if(
       cmdline.isset("dump-c") || cmdline.isset("dump-cpp") ||
+      cmdline.isset("create-function-stub") ||
       cmdline.isset("dump-c-type-header"))
     {
       const bool is_cpp=cmdline.isset("dump-cpp");
+      const bool c_stub = cmdline.isset("create-function-stub");
       const bool is_header = cmdline.isset("dump-c-type-header");
       const bool h_libc=!cmdline.isset("no-system-headers");
       const bool h_all=cmdline.isset("use-all-headers");
       const bool harness=cmdline.isset("harness");
       namespacet ns(goto_model.symbol_table);
 
+      if(c_stub)
+      {
+        bool okay = true;
+        for(const char *flag : {"dump-c",
+                                "dump-cpp",
+                                "dump-c-type-header",
+                                "no-system-headers",
+                                "use-all-headers",
+                                "harness"})
+        {
+          if(cmdline.isset(flag))
+          {
+            okay = false;
+            log.error() << "--create-function-stub cannot be used with --"
+                        << flag << messaget::eom;
```

please throw a `invalid_command_line_argument_exceptiont` instead

---

thanks